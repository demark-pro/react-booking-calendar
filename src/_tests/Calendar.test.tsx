import React from "react";
import { render } from "@testing-library/react";
import { Calendar } from "..";

describe("Calendar", () => {
  test("renders the Calendar component", () => {
    render(<Calendar selected={[new Date(), null]} />);
  });
});
