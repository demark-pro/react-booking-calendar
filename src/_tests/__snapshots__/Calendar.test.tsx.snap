// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - Calendar 1`] = `
.emotion-0 {
  background-color: #fff;
  position: relative;
  height: 100%;
  width: 100%;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  -webkit-font-smoothing: antialiased;
  box-sizing: border-box;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0.75rem 0.25rem;
  position: relative;
  box-sizing: border-box;
}

.emotion-2 {
  -webkit-transform: rotate(0);
  -moz-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
  fill: #007aff;
  width: 28px;
  height: 28px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  box-sizing: border-box;
}

.emotion-2:hover {
  background: #eef5ff;
  border-radius: 50%;
}

.emotion-3 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  font-size: 16px;
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  box-sizing: border-box;
}

.emotion-4 {
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  fill: #007aff;
  width: 28px;
  height: 28px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  box-sizing: border-box;
}

.emotion-4:hover {
  background: #eef5ff;
  border-radius: 50%;
}

.emotion-5 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  left: 0;
  right: 0;
  background: #fff;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  width: 100%;
  box-sizing: border-box;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: calc(100% / 7);
  -ms-flex-preferred-size: calc(100% / 7);
  flex-basis: calc(100% / 7);
  width: calc(100% / 7);
  font-size: 10px;
  text-transform: uppercase;
  padding: 0.5rem 0;
  font-weight: 700;
  color: red;
  box-sizing: border-box;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: calc(100% / 7);
  -ms-flex-preferred-size: calc(100% / 7);
  flex-basis: calc(100% / 7);
  width: calc(100% / 7);
  font-size: 10px;
  text-transform: uppercase;
  padding: 0.5rem 0;
  font-weight: 700;
  color: #424242;
  box-sizing: border-box;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 0.75rem;
  box-sizing: border-box;
}

.emotion-14 {
  height: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  list-style: none;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
  cursor: pointer;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  width: calc(100%/7);
  -webkit-flex-basis: calc(100%/7);
  -ms-flex-preferred-size: calc(100%/7);
  flex-basis: calc(100%/7);
  margin: 0.15rem 0;
  box-sizing: border-box;
}

.emotion-14:hover {
  cursor: default;
}

.emotion-15 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  top: 6px;
  font-size: 9px;
  color: #969696;
  box-sizing: border-box;
}

.emotion-16 {
  z-index: 3;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #cccccc;
  box-sizing: border-box;
}

.emotion-17 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  bottom: 3px;
  font-size: 8px;
  color: #969696;
  box-sizing: border-box;
}

.emotion-114 {
  height: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  list-style: none;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
  cursor: pointer;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  width: calc(100%/7);
  -webkit-flex-basis: calc(100%/7);
  -ms-flex-preferred-size: calc(100%/7);
  flex-basis: calc(100%/7);
  margin: 0.15rem 0;
  box-sizing: border-box;
}

.emotion-114:hover {
  background: #eef5ff;
  border-radius: 6px;
}

.emotion-115 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  top: 6px;
  font-size: 9px;
  color: #0d6efd;
  box-sizing: border-box;
}

.emotion-116 {
  z-index: 3;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0d6efd;
  box-sizing: border-box;
}

.emotion-117 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  bottom: 3px;
  font-size: 8px;
  color: #0d6efd;
  box-sizing: border-box;
}

.emotion-119 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  top: 6px;
  font-size: 9px;
  color: #424242;
  box-sizing: border-box;
}

.emotion-120 {
  z-index: 3;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #424242;
  box-sizing: border-box;
}

.emotion-121 {
  position: absolute;
  left: 0;
  right: 0;
  line-height: 12px;
  text-transform: capitalize;
  z-index: 3;
  bottom: 3px;
  font-size: 8px;
  color: #424242;
  box-sizing: border-box;
}

<div>
  <div
    class="calendar__calendar-container emotion-0"
  >
    <div
      class="calendar__month-container emotion-1"
    >
      <svg
        class="calendar__month-arrowBack emotion-2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 17a1 1 0 0 1-.707-.293l-4-4a1 1 0 0 1 0-1.414l4-4a1 1 0 1 1 1.414 1.414L11.414 12l3.293 3.293A1 1 0 0 1 14 17z"
          style="fill: #007aff;"
        />
      </svg>
      <div
        class="calendar__month-content emotion-3"
      >
        <span
          style="color: rgb(0, 122, 255);"
        >
          February
        </span>
        <span
          style="color: rgb(0, 122, 255);"
        >
          2023
        </span>
      </div>
      <svg
        class="calendar__month-arrowNext emotion-4"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 17a1 1 0 0 1-.707-.293l-4-4a1 1 0 0 1 0-1.414l4-4a1 1 0 1 1 1.414 1.414L11.414 12l3.293 3.293A1 1 0 0 1 14 17z"
          style="fill: #007aff;"
        />
      </svg>
    </div>
    <div
      class="calendar__week-container emotion-5"
    >
      <div
        class="calendar__weekCell emotion-6"
      >
        Su
      </div>
      <div
        class="calendar__weekCell emotion-7"
      >
        Mo
      </div>
      <div
        class="calendar__weekCell emotion-7"
      >
        Tu
      </div>
      <div
        class="calendar__weekCell emotion-7"
      >
        We
      </div>
      <div
        class="calendar__weekCell emotion-7"
      >
        Th
      </div>
      <div
        class="calendar__weekCell emotion-7"
      >
        Fr
      </div>
      <div
        class="calendar__weekCell emotion-6"
      >
        Sa
      </div>
    </div>
    <div
      class="calendar__day-container emotion-13"
    >
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          29
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          30
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        >
          Jan
        </div>
        <div
          class="calendar__dayCell-content emotion-16"
        >
          31
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          1
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          2
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          3
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          4
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          5
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          6
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          7
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          8
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          9
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          10
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          11
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          12
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          13
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          14
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          15
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          16
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          17
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          18
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          19
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          20
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          21
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-14"
      >
        <div
          class="calendar__dayCell-header emotion-15"
        />
        <div
          class="calendar__dayCell-content emotion-16"
        >
          22
        </div>
        <div
          class="calendar__dayCell-footer emotion-17"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-115"
        >
          today
        </div>
        <div
          class="calendar__dayCell-content emotion-116"
        >
          23
        </div>
        <div
          class="calendar__dayCell-footer emotion-117"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          24
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          25
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          26
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          27
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          28
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        >
          Mar
        </div>
        <div
          class="calendar__dayCell-content emotion-120"
        >
          1
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          2
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          3
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
      <div
        class="calendar__dayCell emotion-114"
      >
        <div
          class="calendar__dayCell-header emotion-119"
        />
        <div
          class="calendar__dayCell-content emotion-120"
        >
          4
        </div>
        <div
          class="calendar__dayCell-footer emotion-121"
        />
      </div>
    </div>
  </div>
</div>
`;
